staging/src/k8s.io/apiserver/pkg/storage/etcd3/watcher.go

```diff
@@ -462,9 +451,6 @@ func decodeObj(codec runtime.Codec, versioner storage.Versioner, data []byte, re
 	obj, err := runtime.Decode(codec, []byte(data))
 	if err != nil {
 		if fatalOnDecodeError {
```

Comment:
Shouldn't we remove this entire panic behavior, since there's no special transformer that produces the magic value anymore?

Reply:
It wasn't the magic behavior that was causing panics - we were just catching a special type of error.
I removed that one.

Regarding the "fatalOnDecodeError" - it was originally added to make sure that any errors will cause tests to panic and won't have any chance to be magically hidden (as not affecting what exactly is being tested). I think that usecase is potentially still valid.
[My personal opinion is that it's no longer needed - but I think removing that is a separate thing - I would be happy to file a separate PR for it, but let's not tight that to partially unrelated thing of a special kind of error.

Reply:
ACK, I see the justification now in https://github.com/kubernetes/kubernetes/pull/46045 - didn't think through the colocated resources case.